src/wasm/wasm-binary.cpp

```diff
@@ -486,14 +489,49 @@ void WasmBinaryWriter::writeLateUserSections() {
 }
 
 void WasmBinaryWriter::writeUserSection(const UserSection& section) {
-  auto start = startSection(0);
+  auto start = startSection(BinaryConsts::User);
   writeInlineString(section.name.c_str());
   for (size_t i = 0; i < section.data.size(); i++) {
     o << uint8_t(section.data[i]);
   }
   finishSection(start);
 }
 
+void WasmBinaryWriter::writeFeaturesSection() {
+  if (!wasm->emitFeaturesSection || wasm->features <= FeatureSet{}) {
+    return;
+  }
+
+  // TODO(tlively): unify feature names with rest of toolchain and use
```

Comment:
i don't understand this TODO - are these not already in harmony with the rest of the toolchain? What would need to change?

Reply:
the names used here are different from the names in wasm-features.h. For example "threads" is called "atomics" in the rest of the toolchain. There are other differences as well.